vyper/parser/stmt.py

```diff
@@ -177,16 +183,29 @@ def assign(self):
             raise StructureException("Assignment statement must have one target", self.stmt)
         self.context.set_in_assignment(True)
         sub = Expr(self.stmt.value, self.context).lll_node
+
+        # Not sure if we should go for assignments or if there
+        # is a more general place to check for this.
+        # if isinstance(sub.typ, StructType) and sub.typ.name is None :
+        #     warnings.warn(
+        #             "Anonymous struct values have been removed in"
+        #             " favor of named structs, see VIP300",
+        #             DeprecationWarning
+        #             )
```

Comment:
I think this can be moved to global_context.py around line 82 / AnnAssign?

Reply:
Actually if we're removing them entirely I can move the check upstream into `expr.py`

Reply:
Fixed in 9e7bdac9fef9c737b028b223fcdf0d7f715ede82